src/service/OAuth.service.ts

```diff
@@ -0,0 +1,70 @@
+import { Fetch } from './Fetch';
+import LAMP from '../index';
+
+export interface OAuthParams {
+  codeVerifier?: string
+  codeChallenge?: string
+}
+
+export interface StartFlowResponse {
+  loginURL?: URL
+  logoutURL?: URL
+}
+
+export interface AuthResponse {
+  success: boolean | undefined,
+  access_token: string | undefined,
+  refresh_token: string | undefined,
+}
+export class OAuthService {
```

Comment:
The sessionStorage stuff here seems more for the dashboard codebase instead? The `LAMP-js` library needs to be usable if a data scientist wants to use NodeJS for their data analysis/visualization dashboard/etc.